lib/kube/proxy/forwarder.go

```diff
@@ -864,9 +862,9 @@ func (f *Forwarder) join(ctx *authContext, w http.ResponseWriter, req *http.Requ
 		party := newParty(*ctx, stream.Mode, client)
 		go func() {
 			<-stream.Done()
-			session.mu.Lock()
-			defer session.mu.Unlock()
-			session.leave(party.ID)
+			if err := session.leave(party.ID); err != nil {
```

Comment:
Do you need to hold `mu` for `session.leave()`?

Reply:
I moved the lock into the `session.leave` to simplify the call.